azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -133,6 +133,7 @@ def execute(self, database: str, query: str, properties: Optional[ClientRequestP
         :return: Kusto response data set.
         :rtype: azure.kusto.data.response.KustoResponseDataSet
         """
+        self.validate_endpoint()
```

Comment:
You can remove all of these like in the aio client, and only have it in _execute